src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +176,124 @@ public boolean equals(Object o) {
         return true;
     }
 
+    /**
+     * Add an alias to a SAMSequenceRecord. This can be use to provide some
+     * alternate names fo a given contig. e.g:
+     * <code>1,chr1,chr01,01,CM000663,NC_000001.10</code> e.g:
+     * <code>MT,chrM</code>
+     * 
+     * @param originalName
+     *            existing contig name
+     * @param altName
+     *            new contig name
+     * @return the contig associated to the 'originalName/altName'
+     */
+    public SAMSequenceRecord addSequenceAlias(final String originalName,
+            final String altName) {
+        if (originalName == null)
+            throw new IllegalArgumentException("original name cannot be null");
+        if (altName == null)
+            throw new IllegalArgumentException("alt name cannot be null");
+        final SAMSequenceRecord originalSeqRecord = getSequence(originalName);
+        if (originalSeqRecord == null)
+            throw new IllegalArgumentException("Sequence " + originalName
+                    + " doesn't exist in dictionary.");
+        // same name, nothing to do
+        if (originalName.equals(altName))
+            return originalSeqRecord;
+        final SAMSequenceRecord altSeqRecord = getSequence(altName);
+        if (altSeqRecord != null) {
+            // alias was already set to the same record
+            if (altSeqRecord.equals(originalSeqRecord))
+                return originalSeqRecord;
+            // alias was already set to another record
+            throw new IllegalArgumentException("Alias " + altName
+                    + " was already set to " + altSeqRecord.getSequenceName());
+        }
+        mSequenceMap.put(altName, originalSeqRecord);
```

Comment:
I put a new key in `mSequenceMap` but size() gets its values from `mSequences` isn't it ?

```java
    public int size() {
        return mSequences.size();
    }
```